### ISSUE TITLE
Division by zero in software renderer (LinuxKMS on RISC-V 64bit musl)

### ISSUE THREAD

**Issue by sandreas** [slint-ui/slint#11072]
### Bug Description

I somehow managed to get a division by zero error pointing to the following code line:

https://github.com/slint-ui/slint/blob/7b2b2eb3de2c8443c374429b922dcbf9fb29e368/internal/renderers/software/fixed.rs#L146

Unfortunately I did not find a way to reproduce this yet, but still I'd like to report it since this should not happen and the code seems not to handle this properly.

I'll report back if I find a way to reproduce this - despite you find it necessary to provide a fix upfront checking for zero.

Here is another possible division by zero code part:
https://github.com/slint-ui/slint/blob/7b2b2eb3de2c8443c374429b922dcbf9fb29e368/internal/renderers/software/fixed.rs#L162

### Reproducible Code (if applicable)

Not yet

### Environment Details

- Slint Version: Latest
- Platform/OS: RISC-V 64bit musl, Linux
- Programming Language: Rust
- Backend/Renderer: LinuxKMS, Software

### Product Impact

Crash

**Comment by ogoffart** [comment]
Thanks for making a report.
Would you perhaps have a more deep back trace?
Without way to reproduce, it is hard to do anything with this report.

Maybe this could be related to very big or very small images 

(Similar to https://github.com/slint-ui/slint/issues/7863 )

**Comment by sandreas** [comment]
Here is a stacktrace. I've found a reliable way to reproduce this, but the build is optimized for release, so I don't think this will help much. I try to reproduce this with a debug build providing more info asap:

```

thread 'main' (730) panicked at .cargo/git/checkouts/slint-58e5081c4195d215/b51e989/internal/renderers/software/fixed.rs:146:9:
attempt to divide by zero
stack backtrace:
   0:       0x2aba47a19e - <unknown>
   1:       0x2aba168028 - <unknown>
   2:       0x2aba4799b6 - <unknown>
   3:       0x2aba4797ae - <unknown>
   4:       0x2aba497d04 - <unknown>
   5:       0x2aba497c96 - <unknown>
   6:       0x2aba497c86 - <unknown>
   7:       0x2aba014e2a - <unknown>
   8:       0x2aba014ec6 - <unknown>
   9:       0x2aba1eb9a2 - <unknown>
  10:       0x2aba1ece28 - <unknown>
  11:       0x2aba1eba64 - <unknown>
  12:       0x2aba1eebda - <unknown>
  13:       0x2aba1ee79e - <unknown>
  14:       0x2aba200012 - <unknown>
  15:       0x2aba1ffebc - <unknown>
  16:       0x2aba262916 - <unknown>
  17:       0x2aba291396 - <unknown>
  18:       0x2aba077b02 - <unknown>
  19:       0x2aba07d8c8 - <unknown>
  20:       0x2aba291316 - <unknown>
  21:       0x2aba2913f0 - <unknown>
  22:       0x2aba077b02 - <unknown>
  23:       0x2aba07d8dc - <unknown>
  24:       0x2aba079408 - <unknown>
  25:       0x2aba07dbec - <unknown>
  26:       0x2aba291316 - <unknown>
  27:       0x2aba2913f0 - <unknown>
  28:       0x2aba079126 - <unknown>
  29:       0x2aba07dc00 - <unknown>
  30:       0x2aba07a586 - <unknown>
  31:       0x2aba05ba46 - <unknown>
  32:       0x2aba291316 - <unknown>
  33:       0x2aba2913f0 - <unknown>
  34:       0x2aba07a0ba - <unknown>
  35:       0x2aba05ba46 - <unknown>
  36:       0x2aba291316 - <unknown>
  37:       0x2aba2913f0 - <unknown>
  38:       0x2aba07a0ba - <unknown>
  39:       0x2aba05ba46 - <unknown>
  40:       0x2aba291316 - <unknown>
  41:       0x2aba2913f0 - <unknown>
  42:       0x2aba07a0ba - <unknown>
  43:       0x2aba05ba46 - <unknown>
  44:       0x2aba291316 - <unknown>
  45:       0x2aba2913f0 - <unknown>
  46:       0x2aba07a0ba - <unknown>
  47:       0x2aba05ba5a - <unknown>
  48:       0x2aba291316 - <unknown>
  49:       0x2aba291450 - <unknown>
  50:       0x2aba2025ec - <unknown>
  51:       0x2aba2047f6 - <unknown>
  52:       0x2aba204b5c - <unknown>
  53:       0x2aba203ec8 - <unknown>
  54:       0x2aba1e34be - <unknown>
  55:       0x2aba06bd88 - <unknown>
  56:       0x2aba087482 - <unknown>
  57:       0x2aba0fc8e6 - main
  58:       0x3ff11264b6 - libc_start_main_stage2
                               at /opt/thead/toolchain/riscv-musl/src/env/__libc_start_main.c:95:2

```

**Comment by ogoffart** [comment]
With the information i have, i'd say this is a duplicate of https://github.com/slint-ui/slint/issues/7863 , but thanks anyway for reporting.

**Comment by sandreas** [comment]
> With the information i have, i'd say this is a duplicate of https://github.com/slint-ui/slint/issues/7863 , but thanks anyway for reporting.

Quite possibly. I'm going to retest this as soon as I upgraded my branch.
